React/Views/RCTWebView.m

```diff
@@ -248,6 +270,13 @@ - (void)webView:(__unused UIWebView *)webView didFailLoadWithError:(NSError *)er
 
 - (void)webViewDidFinishLoad:(UIWebView *)webView
 {
+  if (_messagingEnabled) {
+    NSString *source = [NSString stringWithFormat:
+      @"window.postMessage = function(message) {"
+      "  window.location = '%@://%@?' + encodeURIComponent(JSON.stringify(message));"
+      "};", RCTJSNavigationScheme, RCTJSPostMessageHost];
```

Comment:
I think even something like this would be fine to avoid overwriting any existing method,

```js
var originalPostMessage = window.postMessage;
window.postMessage = function(message) {
  if (originalPostMessage) originalPostMessage(message);
  window.location = '%@://%@?' + encodeURIComponent(JSON.stringify(message));
}
```